packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -97,4 +98,32 @@ mixin HasPaint<T extends Object> on Component {
   void tint(Color color, {T? paintId}) {
     getPaint(paintId).colorFilter = ColorFilter.mode(color, BlendMode.srcATop);
   }
+
+  @override
+  double get opacity => paint.color.opacity;
+
+  @override
+  set opacity(double value) {
+    paint.color = paint.color.withOpacity(value);
+    for (final paint in _paints.values) {
+      paint.color = paint.color.withOpacity(value);
+    }
+  }
+
+  OpacityProvider opacityProviderForPaint(T paintId) {
+    return _ProxyOpacityProvider(paintId, this);
+  }
+}
+
+class _ProxyOpacityProvider<T extends Object> implements OpacityProvider {
+  _ProxyOpacityProvider(this.paintId, this.target);
+
+  final T paintId;
+  final HasPaint<T> target;
+
+  @override
+  double get opacity => target.getOpacity(paintId: paintId);
+
+  @override
+  set opacity(double value) => target.setOpacity(opacity, paintId: paintId);
```

Comment:
```suggestion
  set opacity(double value) => target.setOpacity(value, paintId: paintId);
```
I guess it means we don't cover this provider tests?

Reply:
Oh!! Good catch! I've updated this and a test to check this. I think we can create an issue for adding unit tests for `HasPaint` mixin.